src/main/java/cpw/mods/jarhandling/impl/Jar.java

```diff
@@ -280,8 +280,8 @@ private List<Provider> gatherProviders(BiPredicate<String, String> filter) {
         if (!Files.exists(services))
             return List.of();
 
-        try {
-            return Files.list(services)
+        try (var servicesDirStream = Files.list(services)) {
```

Comment:
I thought this had issues with Zip file systems, calling close anywhere on them caused the entire thing to shut down.